### PR TITLE
feat:proper badge theme use for server list

### PR DESCRIPTION
**Pull request by rahimrahman**
#### Summary
While fixing issue with https://github.com/mattermost/mattermost-mobile/pull/8312, there has been a few changes to the badge in the server list requested by @asaadmahmood.

#### Checklist

- [ ] Added or updated unit tests (required for all new features)
- [X] Has UI changes
- [ ] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.
- [ ] Have run E2E tests by adding label `E2E iOS tests for PR`.

#### Device Information
This PR was tested on: iPhone 15 Simulator, 17.4.1

#### Screenshots

denim-sapphire-quartz-indigo-onyx-after-asaad-fix-2

#### Release Note

```release-note

```

### CODE REVIEW

File: app/screens/home/channel_list/servers/servers_list/server_item/server_item.tsx [hunk 1]
```diff
@@ -59,7 +59,7 @@ const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
     badge: {
         left: 18,
         top: -5,
-        borderColor: theme.centerChannelBg,
+        borderWidth: 1,
```
Comment (rahimrahman) [inline app/screens/home/channel_list/servers/servers_list/server_item/server_item.tsx:62]: @asaadmahmood your request to make this a 1px, but the figma you sent is 2px.

@larkox I believe we should set up the proper protocol if we want to move from 2px to 1px.

**Comment by rahimrahman** [comment]
For comparison, this is Quartz before fix:

quartz-fixed

**Comment by matthewbirtch** [comment]
@rahimrahman the Denim and Sapphire themes look good in the screenshots. However, Quartz, Onyx, and Indigo aren't quite right. They should not have blue borders surrounding the badge and the badge background and number should follow the button colors. 

It looks like you're using the right theme color variables in the code, but the screenshots aren't showing right. They should look like this:

image

**Comment by enahum** [comment]
> @rahimrahman the Denim and Sapphire themes look good in the screenshots. However, Quartz, Onyx, and Indigo aren't quite right. They should not have blue borders surrounding the badge and the badge background and number should follow the button colors. 
> 
> 
> 
> It looks like you're using the right theme color variables in the code, but the screenshots aren't showing right. They should look like this:
> 
> 
> 
> image
> 
> 

@matthewbirtch I have a question.
Why if the code is exactly the same for all themes, some look ok but others don't?

Cause as far as I can see there is no specific code based on the theme.

**Comment by matthewbirtch** [comment]
> @matthewbirtch I have a question.
> Why if the code is exactly the same for all themes, some look ok but others don't?
> 
> Cause as far as I can see there is no specific code based on the theme.

@enahum @rahimrahman is it possible the screenshots are just old? They don't match with what is applied in code.

My comment was just based on the screenshots, but I can spin up a build and test that to confirm

**Comment by rahimrahman** [comment]
@matthewbirtch sorry, I think I screwed up my screenshot with a different branch.  This is the correct one:

denim-sapphire-quartz-indigo-onyx-after-asaad-fix-2

The 1px borderWidth change requested might need/should get reverted.  It's too skinny IMO.

**Comment by rahimrahman** [comment]
Back to 2px borderWidth

denim-sapphire-quartz-indigo-onyx-after-asaad-fix-2-borderWidth-2px

**Comment by rahimrahman** [comment]
Monokai & Organization:

monokia-organization-after-asaad-fix-2-borderWidth-2px

**Review by asaadmahmood — APPROVED** [review]
Looks good to me!

**Review by larkox — APPROVED** [review]
Codewise, LGTM.(we still need to revert the border width change, though).

**Review by matthewbirtch — APPROVED** [review]
Thanks @rahimrahman. Looks good!

**Comment by rahimrahman** [comment]
@matthewbirtch @asaadmahmood I don't think there's much to test here other than what's already been displayed in a few screenshots.

Should we skip `QA Review`?

**Comment by yasserfaraazkhan** [comment]
should be good to merge. It'll be tested during code freeze.

**Pull request merged**